internal/ethapi/api.go

```diff
@@ -1450,7 +1473,11 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
 		return nil, err
 	}
 	if tx != nil {
-		return newRPCTransaction(tx, blockHash, blockNumber, index), nil
+		block, err := s.b.BlockByHash(ctx, blockHash)
```

Comment:
I think it's enough to do HeaderByHash, that prevents us from loading all the txs and uncles. Would be even nicer if we didn;t do `ByHash` since that loads up a block, rather simply `GetHeader`, but that's not exposed, so, yeah, maybe HeaderByHash will have to do.